src/js/Components/UsernameForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import { Paragraph } from 'grommet';
+import { Next } from 'grommet-icons';
+import Button from '../Components/Button.jsx';
+import TextInput from '../Components/TextInput.jsx';
+
+const UsernameForm = ({ data, send }) => {
+  const [userName, setUsername] = useState('');
+  const [setIsUsernameFormSubmitted] = useState(false);
```

Comment:
We should pass `isUsernameFormSubmitted` and  `setIsUsernameFormSubmitted` as props to this component from the parent component instead of initializing them here. 

There is logic that relies on `isUsernameFormSubmitted` in the parent component.